src/commands/create/utils.ts

```diff
@@ -99,6 +100,63 @@ export const setupTemplate = async (
   packageManager: PackageManagerType
 ) => {
   Logger.info(`\nSetting up template in ${chalk.magentaBright(folderLocation)}...`);
+  const typedTemplate = template as Template;
+  if (isFramework(typedTemplate, "Foundry")) {
+    await setupFoundryProject(template, folderLocation, packageManager);
+    return;
+  }
+
+  await setupHardhatProject(template, folderLocation);
+
+  if (Object.keys(env).length > 0) {
+    await setupEnvironmentVariables(folderLocation, env);
+  }
+
+  await installDependencies(packageManager, folderLocation);
+};
+// Sets up a foundry project by initializing it with the specified template.
+// Primarily only used for foundry quickstart templates.
+const setupFoundryProject = async (
+  template: GenericTemplate,
+  folderLocation: string,
+  packageManager: PackageManagerType
+) => {
+  const spinner = ora("Initializing foundry project...").start();
+  const manager = packageManagers[packageManager];
+  if (!manager) {
+    spinner.fail("Forge usage not detected.");
+    return;
+  }
+  if (await manager.isInstalled()) {
```

Comment:
`if (!(await manager.isInstalled())) throw ...` to not grow this huge tree of `if` statements

Reply:
Please do the same in all other cases as well